### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by qingyu0530** [jaehyun513/STUNT#4]
hi, jaehyun.
When I run other datasets, such as dna datasets, how should the hyperparameters be set to achieve the highest accuracy? Examples include shot, test_num_way, query, and out_features and hidden_sizes in the MLP.

**Comment by qingyu0530** [comment]
In the code dataset.py, about the train part, I know the hyperparameter in Table 6 of the paper, but what about the vla part?

**Comment by jaehyun513** [comment]
Hi!
Shot, test_num_way, query is depicted in Appendix B.
For all the experiments, we used hidden sizes 1024 and out_features 1024.

**Comment by qingyu0530** [comment]
```
        meta_val_dataset = Income(tabular_size = 105,
                                    seed=P.seed,
                                    source='val',
                                    shot=1, 
                                    tasks_per_batch=P.test_batch_size,
                                    test_num_way = 2,
                                    query = 30)
```
sorry,sir. I mean this part of the code. When the data set is val, do the parameters in it need to be modified according to the parameters specified in Appendix B?

**Comment by jaehyun513** [comment]
Oh. For validation set, we used shot=1 for all the datasets. Test_num_way is same as class size of the dataset (e.g., income dataset has 2 classes so test_num_way should be 2). Query is differ among datasets and we tried to use the largest # queries as we can.

**Comment by qingyu0530** [comment]
I'm sorry, I have one more question. In these two pieces of code, if you changed the query in the validation set, then 30 in these two pieces of code also needs to change?These two pieces of code are located in income.py
```
        if self.source == 'val':
            xq = np.reshape(
                xq,
                [self.tasks_per_batch, num_way * 30, self.tabular_size])
```

```
query_idx.append(k_idx[num_val_shot:num_val_shot+30])
```

**Comment by jaehyun513** [comment]
Oh yes! That's correct.

**Comment by qingyu0530** [comment]
Thank you very much for your answer to my question, which has been of great help to me. I have modified the code according to your answer, and the accuracy has improved somewhat, but it has not reached the highest accuracy mentioned in the paper. I now suspect that the code in which I generated the data is still faulty. Could you please provide me with the code to generate data from openml? I want to make corrections to my code according to the code you provided. I'd appreciate it if you could.

**Comment by jaehyun513** [comment]
Thanks for your interest! Please email me, then I will provide the codes for other datasets which you want (but it may take some time). Thanks!

**Comment by qingyu0530** [comment]
I have sent an email to your mailbox. Thank you very much for your help

**Comment by qingyu0530** [comment]
hi, jaehyun.
In terms of index generation, is it based on the classification in pseudo_val_y, and then select num_shot in each class?

**Comment by qingyu0530** [comment]
How should I generate the index file？

**Comment by qingyu0530** [comment]
After reading the paper, I understand the generation of data in this way.
Let's take dna data for example. After I downloaded the data from openml, I used 80 percent of the complete data as train data
Twenty percent as test data.
Twenty percent of the train data is val data, corresponding to xtest and ytest variables, and the remaining eighty percent of the data corresponds to xtrain and ytrain variables.
As for index, my understanding is that according to the type of data in the train data, that is, the value in ytrain, we choose shot from each category as index.
Do I understand this correctly? Here's my code
```
import numpy as np
from sklearn.cluster import KMeans
import openml
import pandas as pd
from sklearn import preprocessing

np.random.seed(0)

id = 40670
dataset = openml.datasets.get_dataset(id)
df_X, y, categorical_indicator, attribute_names = dataset.get_data(
    dataset_format="dataframe", target=dataset.default_target_attribute
)

for index,i in enumerate(categorical_indicator):
    if i:
        fea_mapping = {fea:idx for idx,fea in enumerate(set(df_X.iloc[:,index].values.dropna()))}
        #对categorical列进行映射
        df_X.iloc[:,index] = df_X.iloc[:,index].map(fea_mapping)
from sklearn.preprocessing import LabelEncoder
df_y = np.array(LabelEncoder().fit_transform(y))
df_X = np.array(df_X)

x = df_X
min_max_scaler = preprocessing.MinMaxScaler()
x_scaled = min_max_scaler.fit_transform(x)
df_X = x_scaled

mask = [0]*len(df_y)
mask = np.random.random([len(df_y),1])
known_mask = mask<0.8
known_mask = known_mask.reshape(-1)
x = df_X[known_mask]
y = df_y[known_mask]
np.save('../{}/xtrain.npy'.format(id), x)
np.save('../{}/ytrain.npy'.format(id), y)
# x = np.load('xtrain.npy')
# y = np.load('ytrain.npy')

test_x =df_X[~known_mask]
test_y = df_y[~known_mask]
np.save('../{}/xtest.npy'.format(id), test_x)
np.save('../{}/ytest.npy'.format(id), test_y)

num_train = int(len(x) * 0.8)
idx = np.random.permutation(len(x))
train_idx = idx[:num_train]
val_idx = idx[num_train:]

train_x = x[train_idx]
val_x = x[val_idx]

np.save('../{}/train_x.npy'.format(id), train_x)
np.save('../{}/val_x.npy'.format(id), val_x)

val_y = y[val_idx]
np.save('../{}/val_y.npy'.format(id), val_y)

# model = KMeans(n_clusters = 3)
# model.fit(val_x)
# labels = model.predict(val_x)
# np.save('../{}/val_y.npy'.format(id), labels)

num_way = 3
num_shot = 1

for seed in range(100):
    idx = []
    np.random.seed(seed)
    for i in range(0, 2):
        list = []
        for j in range(0, len(y)):
            if y[j] == i:
                list.append(j)
        idx_in_way = np.random.choice(list,num_shot,replace=False)
        idx.extend(idx_in_way)
    np.save('../{}/index1/train_idx_{}.npy'.format(id, seed), idx_in_way)
```
